File: rust/src/bin/0004.rs
use std::io::prelude::*;
use std::str::FromStr;

fn ans( i : Vec<f64> ) -> (f64,f64) {
    let (a,b,c,d,e,f) = (i[0],i[1],i[2],i[3],i[4],i[5]);

    let x = ( (c*e - b*f) / (a*e - b*d) * 1000.0).round() / 1000.0;
    let y = ( (c*d - a*f) / (b*d - a*e) * 1000.0).round() / 1000.0;

    (x,y)
}

fn main() {
    let reader = std::io::stdin();

    reader.lock().lines().for_each(|line|{
        let d : Vec<f64> = line.unwrap().split_whitespace().map(|x| f64::from_str(x).unwrap()).collect();
        let (x,y) = ans(d);
        println!("{:.3} {:.3}",x,y);
    })
}

File: python/0004.py
import sys

def rnd(x):
    tmp = round(x,3)
    if tmp == -0.0:
        return 0.0
    else:
        return tmp
    

for line in sys.stdin:
    [a,b,c,d,e,f] = list(map(int, line.split()))

    if b == 0:
        x = c / a
    elif e == 0:
        x = f / d
    else:  
        x = ( c / b - f / e ) / ( a / b - d / e )
    
    if a == 0:
        y = c / b
    elif d == 0:
        y = f / e
    else:
        y = ( c / a - f / d ) / ( b / a - e / d )

    xr = rnd(x)
    yr = rnd(y)
    print("{:.3f} {:.3f}".format(xr,yr))
